src/Core/SettingsFields.h

```diff
@@ -545,6 +546,43 @@ void SettingFieldMultiEnum<EnumT, Traits>::readBinary(ReadBuffer & in)
         return std::initializer_list<std::pair<const char*, NEW_NAME>> __VA_ARGS__ .size();\
     }
 
+/* Setting field for specifying user-defined timezone. It is basically a string, but it needs validation.
+ */
+struct SettingFieldTimezone
+{
+    String value;
+    bool changed = false;
+
+    explicit SettingFieldTimezone(std::string_view str = {}) { validateTimezone(std::string(str)); value = str; }
+    explicit SettingFieldTimezone(const String & str) { validateTimezone(str); value = str; }
+    explicit SettingFieldTimezone(String && str) { validateTimezone(std::string(str)); value = std::move(str); }
+    explicit SettingFieldTimezone(const char * str) { validateTimezone(str); value = str; }
+    explicit SettingFieldTimezone(const Field & f) { const String & str = f.safeGet<const String &>(); validateTimezone(str); value = str; }
+
+    SettingFieldTimezone & operator =(std::string_view str) { validateTimezone(std::string(str)); value = str; changed = true; return *this; }
+    SettingFieldTimezone & operator =(const String & str) { *this = std::string_view{str}; return *this; }
+    SettingFieldTimezone & operator =(String && str) { validateTimezone(str); value = std::move(str); changed = true; return *this; }
+    SettingFieldTimezone & operator =(const char * str) { *this = std::string_view{str}; return *this; }
+    SettingFieldTimezone & operator =(const Field & f) { *this = f.safeGet<const String &>(); return *this; }
+
+    operator const String &() const { return value; } /// NOLINT
+    explicit operator Field() const { return value; }
+
+    const String & toString() const { return value; }
+    void parseFromString(const String & str) { *this = str; }
+
+    void writeBinary(WriteBuffer & out) const;
+    void readBinary(ReadBuffer & in);
+
+private:
+    cctz::time_zone validated_tz;
+    void validateTimezone(const std::string & str)
+    {
+        if (str != "" && !cctz::load_time_zone(str, &validated_tz))
```

Comment:
Please check against `extern const char * auto_time_zones[];`, for details see https://github.com/ClickHouse/ClickHouse/blob/master/src/Storages/System/StorageSystemTimeZones.cpp#L7

Reply:
Done, see latest commit